apps/root/src/hooks/useSwapOptions.ts

```diff
@@ -104,7 +104,21 @@ function useSwapOptions(
             });
 
             if (promiseResult.length) {
-              setState({ result: promiseResult, error: undefined, isLoading: false });
+              // If all of them have tx's we dont need to build them
+              if (promiseResult.filter((option) => !option.tx).length === 0) {
+                setState({ result: promiseResult, error: undefined, isLoading: false });
+              } else {
+                await aggregatorService
+                  .buildSwapOptions({ options: promiseResult, recipient: debouncedAccount as Address })
+                  .then((builtResponse) => {
+                    const newResults = promiseResult.map((option) => ({
+                      ...option,
+                      tx: builtResponse[option.swapper.id],
+                    }));
+                    setState({ result: newResults, error: undefined, isLoading: false });
```

Comment:
I think we should exclude from `promiseResult` all quotes that dont are included in builtResponse (no tx)